DEVELOPING_A_RUNTIME.md

```diff
@@ -16,146 +36,161 @@ A **required** exported constant that decides which version of the Runtime API t
+Process ID, which `vercel dev` uses to send the shut down signal to.
+
+> **Hint:** To determine which ephemeral port the child process is listening on,
+> some form of [IPC](https://en.wikipedia.org/wiki/Inter-process_communication) is
+> required. For example, in `@vercel/go` the child process writes the port number
+> to _file descriptor 3_, which is read by the `startDevServer()` function
+> implementation.
+
+It may also return `null` to opt-out of this behavior for a particular request
+path or entrypoint.
 
-- `files`: All source files of the project as a [Files](#files) data structure.
-- `entrypoint`: Name of entrypoint file for this particular build job. Value `files[entrypoint]` is guaranteed to exist and be a valid [File](#files) reference. `entrypoint` is always a discrete file and never a glob, since globs are expanded into separate builds at deployment time.
-- `workPath`: A writable temporary directory where you are encouraged to perform your build process. This directory will be populated with the restored cache from the previous run (if any) for [`analyze`](#analyze) and [`build`](#build).
-- `cachePath`: A writable temporary directory where you can build a cache for the next run. This is only passed to `prepareCache`.
-- `config`: An arbitrary object passed from by the user in the [Build definition](#defining-the-build-step) in `vercel.json`.
+**Example:**
 
-## Examples
+```typescript
+import { spawn } from 'child_process';
+import { StartDevServerOptions } from '@vercel/build-utils';
 
-Check out our [Node.js Runtime](https://github.com/vercel/vercel/tree/master/packages/now-node), [Go Runtime](https://github.com/vercel/vercel/tree/master/packages/now-go), [Python Runtime](https://github.com/vercel/vercel/tree/master/packages/now-python) or [Ruby Runtime](https://github.com/vercel/vercel/tree/master/packages/now-ruby) for examples of how to build one.
+export async function startDevServer(options: StartDevServerOptions) {
+  // Create a child process which will create an HTTP server
+  const child = spawn('my-runtimes-dev-server', [options.entrypoint], {
+    stdio: ['ignore', 'inherit', 'inherit', 'pipe'],
+  });
 
-## Technical Details
+  // In this example, the child process will write the port number to FD 3…
+  const childPort = Number(await bufferStream(child.stdio[3]));
+
+  return { pid: child.pid, port: childPort };
+}
+```
 
 ### Execution Context
 
-A [Serverless Function](https://vercel.com/docs/v2/serverless-functions/introduction) is created where the Runtime logic is executed. The lambda is run using the Node.js 8 runtime. A brand new sandbox is created for each deployment, for security reasons. The sandbox is cleaned up between executions to ensure no lingering temporary files are shared from build to build.
+- Runtimes are executed in a Linux container that closely matches the Servereless Function runtime environment.
+- The Runtime code is executed using Node.js version **12**.
```

Comment:
```suggestion
- The Runtime code is executed using Node.js version **12.x**.
```

Maybe use 12.x to be consistent with our user docs and make it clear that the minor can/will change https://vercel.com/docs/runtimes#official-runtimes/node-js/node-js-version